File: haskell/glue/src/Glue/Lib/IO.hs
module Glue.Lib.IO where

import Glue.Eval (Eval)
import Glue.IR (IR (..))
import Glue.Lib.IO.Print (printFunc, println)
import Glue.Lib.IO.Read (readLine)
import Glue.Module (ModuleInfo, nativeModule)

io :: ModuleInfo Eval
io =
    nativeModule
        "ffi.io"
        [ ("print", NativeFunc printFunc)
        , ("println", NativeFunc println)
        , ("read-line", NativeFunc readLine)
        ]

File: dart/glue/lib/src/lib/io.dart
import 'package:glue/src/lib/io/print.dart';
import 'package:glue/src/lib/io/read.dart';
import 'package:glue/src/module.dart';

/// IO module - input/output functions
/// Mirrors Haskell Glue.Lib.IO exactly

/// The io module containing all IO functions
/// Mirrors Haskell Glue.Lib.IO.io exactly
final ModuleInfo ioModule = nativeModule('ffi.io', [
  ('print', printFunc),
  ('println', println),
  ('read-line', readLine),
]);
